site/src/sphinx/server-annotated-service.rst

```diff
@@ -0,0 +1,725 @@
+
+- Mapping with path prefix, e.g. ``prefix:/hello``
+
+  - a service method will handle every path which starts with the specified prefix.
+
+- Mapping with path variables, e.g ``/hello/{name}`` or ``/hello/:name``
+
+  - a service method will handle the path matched with the specified path pattern. A path variable in the
+    specified pattern may be mapped to a parameter of the service method.
+
+- Mapping with regular expression, e.g. ``regex:^/hello/(?<name>.*)$``
+
+  - a service method will handle the path matched with the specified regular expression. If a named capturing
+    group exists in the regular expression, it may be mapped to a parameter of the service method.
+
+- Mapping with glob pattern, e.g. ``glob:/*/hello/**``
+
+  - a service method will handle the path matched with the specified glob pattern. A wildcard is mapped to
+    an index which starts with ``0``, so it may be mapped to a parameter of the service method.
+
+You can get the value of a path variable, a named capturing group of the regular expression or wildcards of
+the glob pattern in your service method by annotating a parameter with `@Param`_ as follows.
+Please refer to :ref:`parameter-injection` for more information about `@Param`_.
+
+.. code-block:: java
+
+    public class MyAnnotatedServiceClass {
+
+        @Get("/hello/{name}")
+        public HttpResponse pathvar(@Param("name") String name) { ... }
+
+        @Get("regex:^/hello/(?<name>.*)$")
+        public HttpResponse regex(@Param("name") String name) { ... }
+
+        @Get("glob:/*/hello/**")
+        public HttpResponse glob(@Param("0") String prefix, @Param("1") String name) { ... }
+    }
+
+Every service method in the examples so far had a single HTTP method annotation with it. What if you want
+to map more than one HTTP method to your service method? You can use `@Path`_ annotation to specify a path
+and use the HTTP method annotations without a path just to specify the HTTP methods to be mapped as follows.
+
+.. code-block:: java
+
+    public class MyAnnotatedServiceClass {
+        @Get
+        @Post
+        @Put
+        @Delete
+        @Path("/hello")
+        public HttpResponse hello() { ... }
+    }
+
+.. _parameter-injection:
+
+Parameter injection
+-------------------
+
+Let's see the example in the above section again.
+
+.. code-block:: java
+
+    public class MyAnnotatedServiceClass {
+
+        @Get("/hello/{name}")
+        public HttpResponse pathvar(@Param("name") String name) { ... }
+
+        @Get("regex:^/hello/(?<name>.*)$")
+        public HttpResponse regex(@Param("name") String name) { ... }
+    }
+
+A value of a parameter ``name`` is automatically injected as a ``String`` type by Armeria.
+Which types are available for automatic injection? First, the following built-in types are available.
+Armeria tries to convert a value to the specified type and ``IllegalArgumentException`` will be raised
```

Comment:
> Which types are ...? First .... are available. Armeria tries to ...

How about:

Armeria will try to convert the value appropriately if the parameter type is not `String`. `IllegalArgumentException` will be raised if the conversion fails or the parameter type is not one of the following supported types: